needle/modules/storage/data/files_mdm.py

```diff
@@ -0,0 +1,128 @@
+from core.framework.module import BaseModule
+from core.device.device import Device
+from core.utils.menu import choose_from_list_data_protection, choose_from_list, choose_yes_no
+from core.utils.utils import Utils
+from core.utils.printer import Colors
+
+class Module(BaseModule):
+    meta = {
+        'name': 'MDM dump',
+        'author': 'Oliver Simonnet (@mwr)',
+        'description': 'Locate and dump MDM configuration files',
+        'options': (
+            ('bulk', False, False, 'Dump all configuration files in a selected directory.'),
+            ('autosave', False, False, 'Automatically save files. (No save prompt)'),
+            ('output', True, True, 'File output directory.')
+        ),
+    }
+
+    # ==================================================================================================================
+    # UTILS
+    # ==================================================================================================================
+    def __init__(self, params):
+        BaseModule.__init__(self, params)
+        # Setting default output file
+        self.options['output'] = self.local_op.build_output_path_for_file(self, "mdm")
+
+    # Prompt user to select a directory, and return directory string
+    def select_dir(self, directories):
+        option = choose_from_list(directories)
+        if option is 'All': dirs_str =  ' '.join(directories[:-1]) 
+        else: dirs_str = option
+
+        self.printer.info("Selecting files from: [%s%s%s]" % (Colors.B, dirs_str.replace(" ", "\n\t"), Colors.N))
+        return dirs_str
+
+    # Compose cmd string, and return cmd output
+    def get_files(self, dir_str):
+        cmd = '{bin} {dirs_str} -type f -name "*.plist"'.format(bin=self.device.DEVICE_TOOLS['FIND'], dirs_str=dir_str)
+        out = self.device.remote_op.command_blocking(cmd)
+        retrieved_files = out
+
+        return retrieved_files
+
+    # Filter out duplicate directories, and return list
+    def filter_dirs(self, dirs):
+        directories = []; 
+        [directories.append(Utils.get_file_path(x)) for x in dirs if Utils.get_file_path(x) not in directories]
```

Comment:
I would first populate the `directories` array.
Then, to remove duplicates, you can convert it to a `set` and then back to `list`: `list(set(t))`